File: Code-Interview-Case/Ruby/anagram.rb
x=['dog','ear', 'pu', 'egg', 'god', 'up','ogd','gge','aer']
y=Array.new(x)

print ("Array Sebelum :\n")
puts x

for i in 0..8
  y[i]=x[i].chars.sort.join
end
for i in 0..8
  for j in i+1..8
    if y[i] == y[j]
      passingdatax = x[j]
      passingdatay = y[j]
      posisi = j
      for n in posisi.downto(i+1) do
        x[n] = x[n-1]
        y[n] = y[n-1]
      end
      x[i+1] = passingdatax
      y[i+1] = passingdatay
    end
  end
end

print ("Array Sesudah :\n")
puts x

File: Code-Interview-Case/Javascript/anagram.js
/* Anagram Javascript */

var x = ["dog", "ear", "pu", "egg", "god", "up", "ogd", "gge", "aer"];
var y = ["", "", "", "", "", "", "", "", ""];

for (i = 0; i < 9; i++) {
  var sorted = x[i]
    .split("")
    .sort()
    .join("");
  y[i] = sorted;
}
for (var i = 0; i < 9; i++) {
  for (var j = i + 1; j < 9; j++) {
    if (y[i] == y[j]) {
      var passingdatax = x[j];
      var passingdatay = y[j];
      var posisi = j;
      for (var n = posisi; n > i + 1; n--) {
        x[n] = x[n - 1];
        y[n] = y[n - 1];
      }
      x[i + 1] = passingdatax;
      y[i + 1] = passingdatay;
    }
  }
  console.log(x[i]);
}
